backend/src/routes/subscriber.ts

```diff
@@ -0,0 +1,110 @@
+import { PrismaClient } from "@prisma/client/edge";
+import { withAccelerate } from "@prisma/extension-accelerate";
+import { Hono } from "hono";
+import { verify } from "hono/jwt";
+
+export const subscriberRouter = new Hono<{
+  Bindings: {
+    DATABASE_URL: string;
+    JWT_SECRET: string;
+  };
+  Variables: {
+    userId: string;
+  };
+}>();
+
+subscriberRouter.use("/*", async (c, next) => {
+  try {
+    const header = c.req.header("authorization") || "";
+    const token = header && header.split(" ")[1];
+    const user = await verify(token, c.env.JWT_SECRET);
+    if (user) {
+      c.set("userId", user.id);
+      return next();
+    } else {
+      c.status(403);
+      return c.json({ error: "Unauthorized" });
+    }
+  } catch (e) {
+    c.status(403);
+    return c.json({
+      error: "Credentials failed",
+    });
+  }
+});
+
+subscriberRouter.post("/subscribe", async (c) => {
+  const prisma = new PrismaClient({
+    datasourceUrl: c.env.DATABASE_URL,
+  }).$extends(withAccelerate());
```

Comment:
This 
```bash
const prisma = new PrismaClient({
    datasourceUrl: c.env.DATABASE_URL,
  })  
```
can now be re-written as 

```bash
const prisma = getDBInstance(c);
```

Just make sure you 
```bash
import { getDBInstance } from "../db/util";
```

Note: Please change it for all routes in subscriber.ts where db interaction is performed.